### PR TITLE
Toast - new IA

### PR DESCRIPTION
**Pull request by chri2547**
image

### CODE REVIEW

File: _component_design/toast.md [hunk 1]
```diff
@@ -150,26 +166,47 @@ In the informational toast notification example here, the height of the toast in
...
+</section>
+
+<div class="hxRow" markdown="1">
+
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-4-md hxSpan-4-lg" %}
+
+## Variations
 
-### Timing
+You can use the footer to include a [hyperlink]({{site.baseurl}}/style/text-conventions.html) or a [tertiary]({{site.baseurl}}/components/buttons.html#tertiary-button-weight) button CTA.
```
Comment (palindromebart) [inline _component_design/toast.md]: i feel like the word button should also be in the ref text

File: _component_design/toast.md [hunk 2]
```diff
@@ -150,26 +166,47 @@ In the informational toast notification example here, the height of the toast in
...
+
+</div>
+
+</section>
+
+<div class="hxRow" markdown="1">
+
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-4-md hxSpan-4-lg" %}
+
+## Variations
```
Comment (palindromebart) [inline _component_design/toast.md]: should we include an h3 that says "Toast with footer"?

File: _component_design/toast.md [hunk 3]
```diff
@@ -150,26 +166,47 @@ In the informational toast notification example here, the height of the toast in
...
 
 {% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-4-md hxSpan-4-lg" %}
 
-### Placement, animations, and stacking
+### Behaviors
+
+Ensure that toast supports the following behaviors:
+
+ - **Placement and stacking**: Place toasts in the top-right corner of the page, 32px beneath the navigation bar. If more than one toast notifications are triggered simultaneously, vertically stack them 32px apart, with newer toasts placed on top.
+ - **Animation and timing**: The toast fades into place over `400 ms`, remains in place for 10 seconds, and fades out over `400 ms`. If the user hovers over the toast while it is fading out, it immediately gains full opacity. Toasts persist indefinitely on cursor hover. When the cursor moves off the toast, the toast persists for three more seconds, and then fades over the normal `400 ms`.
```
Comment (palindromebart) [inline _component_design/toast.md]: not 100% confident these are the correct names for the functions we use for the animation... checking with ryan...

File: _component_design/toast.md [hunk 4]
```diff
@@ -48,15 +44,20 @@ pagelink: https://rackerlabs.github.io/helix-ui/components/toasts/
...
-- If there's an error and the user can take an action to correct it, use a call to action (CTA).
-- Use complete sentences. If only one sentence is used, do not include a period. However, if there are two or more sentences, use periods.
-- Try not to exceed more than four lines of your message (excluding the CTA).
-- Only use one CTA per toast. Never include “Dismiss” or “Close” as a CTA.
-- Clicking on the “X” closes the toast notification. Hovering over it keeps it open until the cursor moves off of it. (See [Timing]({{page.url}}#timing) section for more detail.)
+- A toast can include an optional call-to-action (CTA) that navigates the user out of the current page.
+- Only use one CTA for a toast. Do not include a **Dismiss** or **Close** CTA.
+- If there is an error and the user can take an action to correct, include a CTA.
+- Use complete sentences when you write the notification message.
+- If the notification message includes only one sentence, do not use a period. **This runs counter to our text conventions**. If there are two or more sentences, use periods.
```
[the remark this answers is not in the document]
Comment (chri2547) [reply _component_design/toast.md]: The old content mentioned to not use a period. I'm going to strike this BP entirely. If it's a sentence, even if just one, use a period.

[the remark this answers is not in the document]
Comment (chri2547) [reply _component_design/toast.md]: Done.

File: _component_design/toast.md [hunk 5]
```diff
@@ -48,15 +44,20 @@ pagelink: https://rackerlabs.github.io/helix-ui/components/toasts/
...
 
-- If there's an error and the user can take an action to correct it, use a call to action (CTA).
-- Use complete sentences. If only one sentence is used, do not include a period. However, if there are two or more sentences, use periods.
-- Try not to exceed more than four lines of your message (excluding the CTA).
-- Only use one CTA per toast. Never include “Dismiss” or “Close” as a CTA.
-- Clicking on the “X” closes the toast notification. Hovering over it keeps it open until the cursor moves off of it. (See [Timing]({{page.url}}#timing) section for more detail.)
+- A toast can include an optional call-to-action (CTA) that navigates the user out of the current page.
+- Only use one CTA for a toast. Do not include a **Dismiss** or **Close** CTA.
+- If there is an error and the user can take an action to correct, include a CTA.
+- Use complete sentences when you write the notification message.
```
[the remark this answers is not in the document]
Comment (chri2547) [reply _component_design/toast.md]: Done.

File: _component_design/toast.md [hunk 6]
```diff
@@ -150,26 +166,47 @@ In the informational toast notification example here, the height of the toast in
...
 <div class="hxRow" markdown="1">
 
 {% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-4-md hxSpan-4-lg" %}
 
-### Placement, animations, and stacking
+### Behaviors
+
+Ensure that toast supports the following behaviors:
+
+ - **Placement and stacking**: Place toasts in the top-right corner of the page, 32px beneath the navigation bar. If more than one toast notifications are triggered simultaneously, vertically stack them 32px apart, with newer toasts placed on top.
```
[the remark this answers is not in the document]
Comment (chri2547) [reply _component_design/toast.md]: Done. Sure, added xref to the eyebrow doc, good idear.

File: _component_design/toast.md (hunk 1 above)
Comment (chri2547) [reply _component_design/toast.md]: good catch, should have been. Done.

File: _component_design/toast.md (hunk 2 above)
Comment (chri2547) [reply _component_design/toast.md]: I don't think so.

**Comment by cyrichardson** [comment]
On it, @chri2547.

File: _component_design/toast.md [hunk 7]
```diff
@@ -66,28 +67,43 @@ pagelink: https://rackerlabs.github.io/helix-ui/components/toasts/
...
 <section class="static-section" markdown="1">
 
+<div class="hxRow" markdown="1">
+
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-8-md hxSpan-8-lg" %}
+
 ## Specifications
 
-<div class="hxRow" markdown="1">
+Refer to the following specifications when you construct toast.
```
Comment (cyrichardson) [inline _component_design/toast.md]: construct a toast.

File: _component_design/toast.md [hunk 8]
```diff
@@ -66,28 +67,43 @@ pagelink: https://rackerlabs.github.io/helix-ui/components/toasts/
...
-Clicking on the “X” closes the toast, but it will go away after approximately 10 seconds if no action is taken.
+<div class="hxRow" markdown="1">
 
-While the height of toasts is dependent on the amount of text, the minimum height is 56px without a CTA. If a CTA or more lines are needed the height can increase to accommodate. Keep notification messages short and succinct, with a maximum of four lines. Use complete sentences since they are less ambiguous.
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-4-md hxSpan-4-lg" %}
 
-Icons are always centered vertically.
+### Composition
+
+Toast is comprised of the following elements:
```
Comment (cyrichardson) [inline _component_design/toast.md]: A toast is

File: _component_design/toast.md [hunk 9]
```diff
@@ -66,28 +67,43 @@ pagelink: https://rackerlabs.github.io/helix-ui/components/toasts/
...
+
+- **Container (required)**: The container holds all elements of the toast notification.
+- **Notification message (required)**: The notification message explains the result of the action taken by the user.
+- **Accent border (required)**: The color of the accent border corresponds to the notification type.
+- **Notification type indicator (required)**: The notification icon corresponds to the notification type.
+- **Close X (required)**: The user can click Close X to close the notification. The notification automatically disappears after 10 seconds.
+- **Footer (optional)**: The footer can contain a link or a button call-to-action (CTA).
```
Comment (cyrichardson) [inline _component_design/toast.md]: a button CTA.

File: _component_design/toast.md [hunk 10]
```diff
@@ -66,28 +67,43 @@ pagelink: https://rackerlabs.github.io/helix-ui/components/toasts/
...
-Icons are always centered vertically.
+### Composition
+
+Toast is comprised of the following elements:
+
+- **Container (required)**: The container holds all elements of the toast notification.
+- **Notification message (required)**: The notification message explains the result of the action taken by the user.
+- **Accent border (required)**: The color of the accent border corresponds to the notification type.
+- **Notification type indicator (required)**: The notification icon corresponds to the notification type.
+- **Close X (required)**: The user can click Close X to close the notification. The notification automatically disappears after 10 seconds.
```
Comment (cyrichardson) [inline _component_design/toast.md]: click **Close X**

File: _component_design/toast.md [hunk 11]
```diff
@@ -98,22 +114,22 @@ Icons are always centered vertically.
...
-
 <div class="hxRow" markdown="1">
 
 {% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-4-md hxSpan-4-lg" %}
 
-### Without a CTA
+### Style
 
-Clicking on the “X” dismisses the toast notification. There are no other actions that can be taken on this type of toast notification.
+There are three types of toast notifications: Success/Confirmation, Informational, and Error.
```
Comment (cyrichardson) [inline _component_design/toast.md]: The three types of toast notifications are Success/Confirmation, Informational, and Error.

File: _component_design/toast.md [hunk 12]
```diff
@@ -98,22 +114,22 @@ Icons are always centered vertically.
...
+
+Use these specifications when you construct toast.
```
Comment (cyrichardson) [inline _component_design/toast.md]: construct a toast.

File: _component_design/toast.md [hunk 13]
```diff
@@ -150,26 +166,47 @@ In the informational toast notification example here, the height of the toast in
...
-## Behaviors
-
 <div class="hxRow" markdown="1">
 
 {% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-4-md hxSpan-4-lg" %}
 
-### Placement, animations, and stacking
+### Behaviors
+
+Ensure that toast supports the following behaviors:
```
Comment (cyrichardson) [inline _component_design/toast.md]: that a toast

File: _component_design/toast.md (hunk 6 above)
Comment (cyrichardson) [inline _component_design/toast.md]: more than one toast notification is triggered

File: _component_design/toast.md [hunk 14]
```diff
@@ -150,26 +166,47 @@ In the informational toast notification example here, the height of the toast in
...
+
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-4-md hxSpan-4-lg" %}
+
+## Variations
 
-### Timing
+You can use the footer to include a [hyperlink]({{site.baseurl}}/style/text-conventions.html) or a [tertiary]({{site.baseurl}}/components/buttons.html#tertiary-button-weight) button CTA.
 
-The toast fades into place over `400 ms`, sits in place for 8 s seconds, and fades out over `400 ms`. If the user hovers over the toast while it is fading out, it immediately gains full opacity. Toasts will persist indefinitely on cursor hover. Upon cursor removal the toast persists for three more seconds then fades over the normal `400 ms`.
+If you include a footer, use these spacing guidelines.
```
Comment (cyrichardson) [inline _component_design/toast.md]: guidelines:

**Comment by cyrichardson** [comment]
@chri2547 Suggestions included inline.

File: _component_design/toast.md [hunk 15]
```diff
@@ -4,11 +4,11 @@ parent: Notifications
 layout: component
 category: Components
 usage: >
-  Toasts are a specialized notification component used to provide feedback on the status of an actions taking place in the background. Toasts are meant to be ephemeral and capture users attention by having the highest Z-index of all components.
+   A toast is a notification component that contains a short message about the status of a user action. Toasts appears on the screen for 10 seconds and then disappears on its own.
```
[the remark this answers is not in the document]
Comment (chri2547) [reply _component_design/toast.md]: Done.

File: _component_design/toast.md [hunk 16]
```diff
@@ -18,25 +18,21 @@ pagelink: https://rackerlabs.github.io/helix-ui/components/toasts/
...
-{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-4-md hxSpan-4-lg" %}
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-8-md hxSpan-8-lg" %}
 
-### When to use
+## When to use
 
-- A toast notification provides information that a user can act upon. Toasts incorporate an optional tertiary CTA that navigate the user out of the current page or context.
-- Do **not** use this pattern for actions that are in context to the page or region - use an [alert bar]({{site.baseurl}}/components/alert-bar.html).
-- Do **not** use a toast notification for promotional or marketing material such as the announcement of new features.
+Use toast when you want to notify the user that a system operation has occurred. For example, use toast when you want to:
```
[the remark this answers is not in the document]
Comment (chri2547) [reply _component_design/toast.md]: Done.

File: _component_design/toast.md [hunk 17]
```diff
@@ -18,25 +18,21 @@ pagelink: https://rackerlabs.github.io/helix-ui/components/toasts/
...
+- Notify the user that an operation resulted in an error.
+- Provide information to the user about an operation.
 
-{% column right:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-8-md hxSpan-8-lg" %}
+Do not use toast to notify the user of promotional or marketing material, such as the announcement of a new feature.
 
-{% figure [caption:"The toast notification component"] [class:"image bg-light border"] %}
-<embed src="{{site.baseurl}}/assets/images/components/notifications/toasts/toast-hero.svg"/>
-{% endfigure %}
+Do not use toast for actions that are in context to the page or region. For those cases, use an [alert bar]({{site.baseurl}}/components/alert-bar.html).
```
[the remark this answers is not in the document]
Comment (chri2547) [reply _component_design/toast.md]: Done.

File: _component_design/toast.md [hunk 18]
```diff
@@ -48,15 +44,20 @@ pagelink: https://rackerlabs.github.io/helix-ui/components/toasts/
 
 <div class="hxRow" markdown="1">
 
-{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-12-md hxSpan-12-lg" %}
+{% column left:"hxCol hxSpan-8-xs hxSpan-12-sm hxSpan-8-md hxSpan-8-lg" %}
+
+## Best Practices
```
[the remark this answers is not in the document]
Comment (chri2547) [reply _component_design/toast.md]: Done.

File: _component_design/toast.md (hunk 11 above)
Comment (chri2547) [reply _component_design/toast.md]: Done.

File: _component_design/toast.md (hunk 12 above)
Comment (chri2547) [reply _component_design/toast.md]: Done.

File: _component_design/toast.md (hunk 13 above)
Comment (chri2547) [reply _component_design/toast.md]: Done.

File: _component_design/toast.md (hunk 6 above)
Comment (chri2547) [reply _component_design/toast.md]: Done.

File: _component_design/toast.md (hunk 14 above)
Comment (chri2547) [reply _component_design/toast.md]: Not including.

File: _component_design/toast.md [hunk 19]
```diff
@@ -4,11 +4,11 @@ parent: Notifications
 layout: component
 category: Components
 usage: >
-  Toasts are a specialized notification component used to provide feedback on the status of an actions taking place in the background. Toasts are meant to be ephemeral and capture users attention by having the highest Z-index of all components.
+   A toast is a notification component that contains a short message about the status of a user action. Toasts appears on the screen for 10 seconds and then disappears.
```
Comment (palindromebart) [inline _component_design/toast.md]: taking notes here: 
toast allows user to navigate away from a process while it happens and the user is notified in their current context when the process completes (or or fails).

the phrase "user initiated process" seems like a useful one here.

please add: 
do not use a toast for form validation.
do not use a toast if a users acknowledgement of the message is required .

File: _component_design/toast.md [hunk 20]
```diff
@@ -48,15 +44,19 @@ pagelink: https://rackerlabs.github.io/helix-ui/components/toasts/
...
 
-### Best Practices
+Consider the following best practices when you construct toast:
 
-- If there's an error and the user can take an action to correct it, use a call to action (CTA).
-- Use complete sentences. If only one sentence is used, do not include a period. However, if there are two or more sentences, use periods.
-- Try not to exceed more than four lines of your message (excluding the CTA).
-- Only use one CTA per toast. Never include “Dismiss” or “Close” as a CTA.
-- Clicking on the “X” closes the toast notification. Hovering over it keeps it open until the cursor moves off of it. (See [Timing]({{page.url}}#timing) section for more detail.)
+- A toast can include an optional call-to-action (CTA) that navigates the user out of the current page, or that triggers an action.
```
Comment (palindromebart) [inline _component_design/toast.md]: Maybe focus on how this is an edge case.
While most toasts do not need a CTA, you may want to include one. 
reasons to include a CTA:
- user has navigated away from the page where the process what initiated and needs to navigate easily back to continue with the job to be done.  E.g. user created a new user, if the process took some time, and they navigated away, they may want to navigate back to that user to configure their permission settings. in this case, a link that navigates the the new user's details page may be useful.
- if the process failed it's may be helpful to include a CTA to retry the process

File: _component_design/toast.md (hunk 19 above)
[the remark this answers is not in the document]
Comment (chri2547) [reply _component_design/toast.md]: Done.

File: _component_design/toast.md [hunk 21]
```diff
@@ -18,25 +18,21 @@ pagelink: https://rackerlabs.github.io/helix-ui/components/toasts/
...
 
-- A toast notification provides information that a user can act upon. Toasts incorporate an optional tertiary CTA that navigate the user out of the current page or context.
-- Do **not** use this pattern for actions that are in context to the page or region - use an [alert bar]({{site.baseurl}}/components/alert-bar.html).
-- Do **not** use a toast notification for promotional or marketing material such as the announcement of new features.
+Use toast when you want to notify the user that a system operation has occurred. For example, use toast when you want to:
 
-{% endcolumn %}
+- Notify the user that an operation was successful.
+- Notify the user that an operation resulted in an error.
+- Provide information to the user about an operation.
```
[the remark this answers is not in the document]
Comment (chri2547) [reply _component_design/toast.md]: Done.

File: _component_design/toast.md (hunk 19 above)
Comment (chri2547) [reply _component_design/toast.md]: Done.

File: _component_design/toast.md (hunk 20 above)
Comment (chri2547) [reply _component_design/toast.md]: Done.

File: _component_design/toast.md [hunk 22]
```diff
@@ -18,25 +18,28 @@ pagelink: https://rackerlabs.github.io/helix-ui/components/toasts/
...
-{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-4-md hxSpan-4-lg" %}
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-8-md hxSpan-8-lg" %}
 
-### When to use
+## When to use
 
-- A toast notification provides information that a user can act upon. Toasts incorporate an optional tertiary CTA that navigate the user out of the current page or context.
-- Do **not** use this pattern for actions that are in context to the page or region - use an [alert bar]({{site.baseurl}}/components/alert-bar.html).
-- Do **not** use a toast notification for promotional or marketing material such as the announcement of new features.
+Users want to know if an action they initiated was successful or not successful. It a user action is not successful, they user wants to know why. Use a toast notification when you want to close the workflow feedback loop on a user action.
```
Comment (palindromebart) [inline _component_design/toast.md]: s/It/If

Comment (palindromebart) [reply _component_design/toast.md]: S/they user/the user

File: _component_design/toast.md [hunk 23]
```diff
@@ -48,15 +51,20 @@ pagelink: https://rackerlabs.github.io/helix-ui/components/toasts/
...
-- Clicking on the “X” closes the toast notification. Hovering over it keeps it open until the cursor moves off of it. (See [Timing]({{page.url}}#timing) section for more detail.)
+- Construct the toast so that the user can navigate away and still be notified in their current context when the user-initiated process completes or fails.
+- A toast can include an optional call-to-action (CTA) that navigates the user out of the current page, or that triggers an action.
+- Only use one CTA for a toast. Do not include a **Dismiss** or **Close** CTA.
+- If there is an error and the user can take an action to correct, include a CTA.
+- Use human-friendly, complete sentences when you write the notification message. For more information about writing notification message, see [Alert and Message Guidelines]({{site.baseurl}}/style/alert-messages.html).
+- The notification message should not exceed more than four lines, excluding the CTA.
+- The user can click close **X** to close the toast.
+- A toast remains open when the user hovers over it and closes when the cursor moves off of it. Refer to [Behaviors]({{page.url}}#behaviors) for more information about toast timing.
```
Comment (palindromebart) [inline _component_design/toast.md]: off of it, provided the full 10 seconds has elapsed.

File: _component_design/toast.md (hunk 22 above)
Comment (chri2547) [reply _component_design/toast.md]: Good catches. Done.

File: _component_design/toast.md (hunk 23 above)
Comment (chri2547) [reply _component_design/toast.md]: Done.

File: _component_design/toast.md [hunk 24]
```diff
@@ -48,15 +44,20 @@ pagelink: https://rackerlabs.github.io/helix-ui/components/toasts/
 
 <div class="hxRow" markdown="1">
 
-{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-12-md hxSpan-12-lg" %}
+{% column left:"hxCol hxSpan-8-xs hxSpan-12-sm hxSpan-8-md hxSpan-8-lg" %}
+
+## Best Practices
 
-### Best Practices
+Consider the following best practices when you construct toast:
```
[the remark this answers is not in the document]
Comment (nimbinatus) [reply _component_design/toast.md]: @chri2547 ^^ not addressed

File: _component_design/toast.md (hunk 10 above)
Comment (nimbinatus) [reply _component_design/toast.md]: @chri2547 ^^ not addressed

File: _component_design/toast.md [hunk 25]
```diff
@@ -48,15 +44,20 @@ pagelink: https://rackerlabs.github.io/helix-ui/components/toasts/
...
-- Try not to exceed more than four lines of your message (excluding the CTA).
-- Only use one CTA per toast. Never include “Dismiss” or “Close” as a CTA.
-- Clicking on the “X” closes the toast notification. Hovering over it keeps it open until the cursor moves off of it. (See [Timing]({{page.url}}#timing) section for more detail.)
+- A toast can include an optional call-to-action (CTA) that navigates the user out of the current page.
+- Only use one CTA for a toast. Do not include a **Dismiss** or **Close** CTA.
+- If there is an error and the user can take an action to correct, include a CTA.
+- Use complete sentences when you write the notification message.
+- If the notification message includes only one sentence, do not use a period. **This runs counter to our text conventions**. If there are two or more sentences, use periods.
+- The notification message should not exceed more than four lines, excluding the CTA.
+- The user can click close **X** to close the toast.
```
[the remark this answers is not in the document]
Comment (nimbinatus) [reply _component_design/toast.md]: @chri2547 ^^ not addressed

File: _component_design/toast.md [hunk 26]
```diff
@@ -48,15 +44,20 @@ pagelink: https://rackerlabs.github.io/helix-ui/components/toasts/
...
+Consider the following best practices when you construct toast:
 
-- If there's an error and the user can take an action to correct it, use a call to action (CTA).
-- Use complete sentences. If only one sentence is used, do not include a period. However, if there are two or more sentences, use periods.
-- Try not to exceed more than four lines of your message (excluding the CTA).
-- Only use one CTA per toast. Never include “Dismiss” or “Close” as a CTA.
-- Clicking on the “X” closes the toast notification. Hovering over it keeps it open until the cursor moves off of it. (See [Timing]({{page.url}}#timing) section for more detail.)
+- A toast can include an optional call-to-action (CTA) that navigates the user out of the current page.
+- Only use one CTA for a toast. Do not include a **Dismiss** or **Close** CTA.
+- If there is an error and the user can take an action to correct, include a CTA.
```
[the remark this answers is not in the document]
Comment (nimbinatus) [reply _component_design/toast.md]: @chri2547 ^^ not addressed

File: _component_design/toast.md (hunk 24 above)
[the remark this answers is not in the document]
Comment (chri2547) [reply _component_design/toast.md]: Done. Missed that one. Thx!

File: _component_design/toast.md (hunk 26 above)
[the remark this answers is not in the document]
Comment (chri2547) [reply _component_design/toast.md]: Weird, missed this one too! Done.

File: _component_design/toast.md (hunk 25 above)
[the remark this answers is not in the document]
Comment (chri2547) [reply _component_design/toast.md]: Done.

File: _component_design/toast.md (hunk 10 above)
Comment (chri2547) [reply _component_design/toast.md]: Done.

**Pull request merged**